libs/sarif/src/schema/multiformat_message_string.rs

```diff
@@ -0,0 +1,129 @@
+use alloc::borrow::Cow;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::schema::PropertyBag;
+
+/// A message string or message format string rendered in multiple formats.
+///
+/// It groups together all available textual formats for a message string
+///
+/// See the [SARIF specification §3.12][spec] for more information.
+///
+/// [spec]: https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317473.
+///
+/// ## Localization
+///
+/// Certain `MultiformatMessageString`-value properties, for example
+/// [`ReportingDescriptor::short_description`], can be translated into other languages. These
+/// properties are described as "localizable". The description of every localizable property will
+/// state that it is localizable.
+///
+/// [`ReportingDescriptor::short_description`]: crate::schema::ReportingDescriptor::short_description
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "camelCase", deny_unknown_fields)
+)]
+pub struct MultiformatMessageString<'s> {
+    /// A plain text representation of the message.
+    ///
+    /// ## Note
+    ///
+    /// This property is required to ensure that the message is viewable even in contexts that do
+    /// not support the rendering of formatted text
+    ///
+    /// See the [SARIF specification §3.12.3][spec] for more information.
+    ///
+    /// [spec]: https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317476
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub text: Cow<'s, str>,
```

Comment:
I am unsure how we want to ensure this, but the spec says it is required to be non-empty.

Reply:
I have noticed this, too, however, the JSON Schema does not require this. A non-empty string makes sense but I don't know if and how we want to ensure this. In my opinion, we should ignore the specs here for now and rely on the Schema for now, especially, as we don't have any concept of an invalid log, yet.

Reply:
Could you please add a note referring to either this comment or that we don't ensure this yet? Just so we can keep track.

Reply:
Added in ca908d2cff2f364e55d6a65efd7935a11e6bfa3b